capz/run-capz-e2e.sh

```diff
@@ -387,7 +387,7 @@ run_e2e_test() {
             # This will not work in community cluster as this secret is not present (hence we only do it if ENV is set)
             # On the community cluster we will use credential providers to a private registry in azure see:
             # https://github.com/kubernetes-sigs/windows-testing/issues/446
-            export KUBE_TEST_REPO_LIST="$SCRIPT_ROOT/../images/image-repo-list-private-registry"
+            export KUBE_TEST_REPO_LIST="$SCRIPT_ROOT/../images/image-repo-list-private-registry-community"
             ADDITIONAL_E2E_ARGS+=("--docker-config-file=${DOCKER_CONFIG_FILE}")
```

Comment:
We need to drop this line and fix the IF statement above removing `&& -n "${DOCKER_CONFIG_FILE:-""}"`.  

We don't run the `pull.from.private.registry.with.secret` test in the PR here but for testing purposes we could hard code `export GINKGO_FOCUS=pull.from.private.registry.with.secret`` for a run to make sure this is working properly